### PR TITLE
Pass base URI struct seperately to serialization function

### PR DESCRIPTION
**Pull request by JKRhb**
This PR tries to lay the groundwork for passing the base URI from a CoAP packet to the serialization function. As soon as #8 (or a similar solution) is merged, the IP address a `GET /.well-known/wot-thing-description` request was sent to should be able to be determined, extracted from the CoAP packet, and added to the TD. Feedback is very much appreciated :)

### CODE REVIEW

**Comment by Citrullin** [comment]
Why do you want to have another parameter in the function, when you can just update the thing itself?

**Comment by JKRhb** [comment]
@Citrullin Do you think we could merge this into the `wot_gcoap` branch or should something else be implemented/optimized first?

File: sys/net/wot/coap/coap.c [hunk 1]
```diff
@@ -105,6 +98,51 @@ void _wot_td_coap_ser_receiver(const char *c){
...
+static int get_base_ip_address(ipv6_addr_t *res) {
+    const int MAX_ADRESSES_TO_CHECK = 5;
+    netif_t* interface = NULL;
+    ipv6_addr_t local_address = {0};
+    ipv6_addr_t ula_address = {0};
+    bool link_local_found = false;
+    bool ula_found = false;
+    int netres;
+
+    while ((interface = netif_iter(interface)) != NULL) {
```
Comment (JKRhb) [inline sys/net/wot/coap/coap.c:110]: @Citrullin Do you think only the net interface that received the request should be checked for IPaddresses? I think that should be relatively easy to implement (using `sock_aux_local`).

Comment (JKRhb) [reply sys/net/wot/coap/coap.c:110]: I merged the PR now into the `wot_gcoap` branch but this is probably something we should keep in mind.

**Pull request merged**